src/include/functions/delta_scan.hpp

```diff
@@ -57,16 +57,21 @@ struct DeltaSnapshot : public MultiFileList {
     // TODO: How to guarantee we only call this after the filter pushdown?
     void InitializeFiles();
 
+    template <class T>
+    T TryUnpackKernelResult(ffi::ExternResult<T> result) {
+        return KernelUtils::UnpackResult<T>(result, StringUtil::Format("While trying to read from delta table: '%s'", paths[0]));
```

Comment:
Might be better for `UnpackResult` to accept a lambda, to avoid allocating a string on the happy path of every kernel call?